contracts/implementation/PoolCommitter.sol

```diff
@@ -628,28 +628,29 @@ contract PoolCommitter is IPoolCommitter, IPausable, Initializable {
                 emit ExecutedCommitsForInterval(executionTracking._updateIntervalId, burningFee);
                 delete totalPoolCommitments[executionTracking._updateIntervalId];
 
-                // counter overflowing would require an unrealistic number of update intervals
+                // counter overflowing would require an unrealistic number of update intervals to be updated
+                // This wouldn't fit in a block, anyway.
                 unchecked {
                     updateIntervalId += 1;
+                    counter += 1;
                 }
             } else {
                 break;
             }
-            // counter overflowing would require an unrealistic number of update intervals to be updated
-            // This wouldn't fit in a block, anyway.
-            unchecked {
-                counter += 1;
-            }
         }
 
         updateMintingFee(
             PoolSwapLibrary.getPrice(longBalance, executionTracking.longTotalSupply),
             PoolSwapLibrary.getPrice(shortBalance, executionTracking.shortTotalSupply)
         );
 
-        // Subtract counter by 1 to accurately reflect how many update intervals were executed
-        if (block.timestamp >= lastPriceTimestamp + updateInterval * (counter - 1)) {
-            // check if finished
+        // if we maxed out the number of intervals upkept and projected lastPriceTimestamp is more than `updateInterval` seconds ago
+        // it means there are more intervals to upkeep
+        // counter will be MAX_ITERATIONS + 1 if we hit max iterations because of the loop condition `while (counter <= MAX_ITERATIONS)`
+        if (
+            counter > MAX_ITERATIONS &&
+            (block.timestamp - (lastPriceTimestamp + updateInterval * (counter - 1))) > updateInterval
```

Comment:
if counter > MAX_ITERATIONS (can only ever be 16) and 
blockTime - (lastTimestamp + updateInterval * 15) > updateInterval
has alternate forms
blockTime > lastTimestamp + 16 * updateInterval
and 
lastTimestamp < blockTime - 16 * updateInterval
Which is a bit funky

Reply:
actually yeah this checks out

Reply:
This should be safe to wrap with `unchecked {}` ?